extensions/ipynb/src/notebookAttachmentCleaner.ts

```diff
@@ -0,0 +1,250 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+
+class AttachmentCleaner {
+
+	// Format:       { notebookUri : { cellFragment : { filename : { mime : b64 } } } }
+	attachmentCache: { [key: string]: any } = {};
+
+	/**
+	 * take in a NotebookDocumentChangeEvent, and clean the attachment data for the cell(s) that have had their markdown source code changed
+	 * @param e NotebookDocumentChangeEvent from the onDidChangeNotebookDocument listener
+	 */
+	cleanNotebookAttachments(e: vscode.NotebookDocumentChangeEvent) {
+		// return if there are no sources changes in event
+		if (!e.cellChanges.length) {
```

Comment:
[nit] Do you need this check? The loop will never exeucute if `cellChanges` is empty